sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala

```diff
@@ -397,6 +397,13 @@ private[hive] object HiveQl {
   protected def extractDbNameTableName(tableNameParts: Node): (Option[String], String) = {
     val (db, tableName) =
       tableNameParts.getChildren.map { case Token(part, Nil) => cleanIdentifier(part) } match {
+        case Seq(dbNameAndTable) if (dbNameAndTable.split("\\.").length > 2) => {
```

Comment:
Nit:I prefer to ` not equals to 2` since user may pass only `.` as the `dbNameAndTable`, the length is 0. 
Then your case will not throw HiveException.